src/contexts/NewDelivery.tsx

```diff
@@ -0,0 +1,189 @@
+import { createContext, ReactNode, useEffect, useState } from 'react'
+import { SubmitHandler } from 'react-hook-form'
+import useNewDeliverySteps from '@/hooks/useNewDeliverySteps'
+import { NewDeliveryContextValues } from '@/interfaces/newDeliveryContextValues'
+import { useTranslation } from 'react-i18next'
+
+interface NewDeliveryProviderProps {
+  children: ReactNode
+}
+
+export const NewDeliveryContext = createContext<NewDeliveryContextValues>({
+  currentStep: 1,
+  formsData: {},
+  openDialog: false,
+  onCloseDialog: () => {},
+  handleBack: () => {},
+  handleClickOpen: () => {},
+  handleSubmit: () => {},
+  updateStep4Item: () => {},
+  deleteStep4Item: () => {},
+  alertMessage: '',
+  isCompletedMove: false,
+  isExceedQuantity: false
+})
+
+interface MoveGood {
+  type: string // palettes | packages | pieces
+  quantity: number
+  zone: string
+}
+
+export default function NewDeliveryProvider({ children }: NewDeliveryProviderProps) {
+  const { t: translate } = useTranslation()
+  const [currentStep, setCurrentStep] = useState(1)
+  const [formsData, setFormsData] = useState<any>({})
+  const [openDialog, setOpenDialog] = useState(false)
+  const steps = useNewDeliverySteps()
+  const [step3Items, setStep3Items] = useState({ pallets: 0, packages: 0, pieces: 0 })
+  const [step4Items, setStep4Items] = useState<MoveGood[]>([{ type: '', quantity: 0, zone: '' }])
+  const [alertMessage, setAlertMessage] = useState('')
+  const [isCompletedMove, setIsCompletedMove] = useState(false)
+  const [isExceedQuantity, setIsExceedQuantity] = useState(false)
+
+  useEffect(() => {
+    // Delete if condition when the newDeliveryProvider is not in main.tsx
+    // Initial set step3Items
+    if (formsData.goods) {
+      const newStep3Items = { ...step3Items }
+      formsData.goods.map((good: any) => {
+        switch (good.goodTypeStep3) {
+          case 'Палети':
+            newStep3Items.pallets = good.goodQuantityStep3
+            return setStep3Items(newStep3Items)
+          case 'Пакети':
+            newStep3Items.packages = good.goodQuantityStep3
+            return setStep3Items(newStep3Items)
+          case 'Бройки':
+            newStep3Items.pieces = good.goodQuantityStep3
+            return setStep3Items(newStep3Items)
+        }
+      })
+    }
+  }, [formsData])
+
+  useEffect(() => {
```

Comment:
you can also split this `useEffect` into smaller util funtions. For example getting current items:
```ts
export function getGoodsItems(items: MoveGood[]) {
     const currentItems = { pallets: 0, packages: 0, pieces: 0 }

    items.map((item) => {
      if (item.type === translate('newDelivery.goodType.pallets')) {
        currentItems.pallets += item.quantity
      } else if (item.type === translate('newDelivery.goodType.packages')) {
        currentItems.packages += item.quantity
      } else if (item.type === translate('newDelivery.goodType.pieces')) {
        currentItems.pieces += item.quantity
      }
      return item
    })

   return currentItems
}
```
This way you have small functions making it easy to understand the `useEffect` on what its doing and also having smaller functions like this one can help writing unit tests for it (if necessary), but also separating it from everything else making it easy to understand on whats doing.

Reply:
You can maybe even do it for the alert message, you can have like a function called `getLeftItemsAlert` or something better, which will return a string (or array which relates to a comment above with making alert messages into an array).

Reply:
Fixed